Packet++/src/GtpLayer.cpp

```diff
@@ -155,522 +155,521 @@ GtpV1Layer::GtpExtension GtpV1Layer::GtpExtension::createGtpExtension(uint8_t* d
 }
 
 
 void GtpV1Layer::parseNextLayer()
 {
-    size_t headerLen = getHeaderLen();
-    if (headerLen < sizeof(gtpv1_header))
-    {
-        // do nothing
-        return;
-    }
-
-    gtpv1_header* header = getHeader();
-    if (header->messageType != PCPP_GTP_V1_GPDU_MESSAGE_TYPE)
-    {
-        // this is a GTP-C message, hence it is the last layer
-        return;
-    }
-
-    if (m_DataLen <= headerLen)
-    {
-        // no data beyond headerLen, nothing to parse further
-        return;
-    }
-
-    // GTP-U message, try to parse the next layer
-
-    uint8_t subProto = *(uint8_t*)(m_Data + headerLen);
-    if (subProto >= 0x45 && subProto <= 0x4e)
-    {
-        m_NextLayer = new IPv4Layer(m_Data + headerLen, m_DataLen - headerLen, this, m_Packet);
-    }
-    else if ((subProto & 0xf0) == 0x60)
-    {
-        m_NextLayer = new IPv6Layer(m_Data + headerLen, m_DataLen - headerLen, this, m_Packet);
-    }
-    else
-    {
-        m_NextLayer = new PayloadLayer(m_Data + headerLen, m_DataLen - headerLen, this, m_Packet);
-    }
+	size_t headerLen = getHeaderLen();
+	if (headerLen < sizeof(gtpv1_header))
+	{
+		// do nothing
+		return;
+	}
+
+	gtpv1_header *header = getHeader();
+	if (header->messageType != PCPP_GTP_V1_GPDU_MESSAGE_TYPE)
+	{
+		// this is a GTP-C message, hence it is the last layer
+		return;
+	}
+
+	if (m_DataLen <= headerLen)
+	{
+		// no data beyond headerLen, nothing to parse further
+		return;
+	}
+
+	// GTP-U message, try to parse the next layer
+
+	uint8_t subProto = *(uint8_t *)(m_Data + headerLen);
+	if (subProto >= 0x45 && subProto <= 0x4e)
+	{
+		m_NextLayer = new IPv4Layer(m_Data + headerLen, m_DataLen - headerLen, this, m_Packet);
+	}
+	else if ((subProto & 0xf0) == 0x60)
+	{
+		m_NextLayer = new IPv6Layer(m_Data + headerLen, m_DataLen - headerLen, this, m_Packet);
+	}
+	else
+	{
+		m_NextLayer = new PayloadLayer(m_Data + headerLen, m_DataLen - headerLen, this, m_Packet);
+	}
 }
 
 size_t GtpV1Layer::getHeaderLen()
 {
-    gtpv1_header* header = getHeader();
-    if (header == NULL)
-    {
-        return 0;
-    }
-
-    size_t res = sizeof(gtpv1_header);
-
-    if (header->messageType != PCPP_GTP_V1_GPDU_MESSAGE_TYPE)
-    {
-        size_t msgLen = be16toh(header->messageLength);
-        res += (msgLen > m_DataLen - sizeof(gtpv1_header) ? m_DataLen - sizeof(gtpv1_header) : msgLen);
-    }
-    else
-    {
-        gtpv1_header_extra* headerExtra = getHeaderExtra();
-        if (headerExtra != NULL && (header->extensionHeaderFlag == 1 || header->sequenceNumberFlag == 1 || header->npduNumberFlag == 1))
-        {
-               res += sizeof(gtpv1_header_extra);
-            GtpExtension nextExt = getNextExtension();
-            while (!nextExt.isNull())
-            {
-                res += nextExt.getTotalLength();
-                nextExt = nextExt.getNextExtension();
-            }
-        }
-    }
-
-    return res;
+	gtpv1_header *header = getHeader();
```

Comment:
please move the `*` back to its original place. This is the convention in the rest of the project

Reply:
Done